### PR TITLE
[WIP] Add documentation for free group, fp groups

### PR DESCRIPTION
**Pull request by gxyd**

### CODE REVIEW

**Comment by gxyd** [comment]
I am of opinion to change the name of module `free_group.py` -> `free_groups.py`, similar to what we have `permutations.py`, `perm_groups.py`, `subsets.py`.

**Comment by jksuom** [comment]
I think that would be fitting. Then we would also have `free_groups.rst`.

File: sympy/combinatorics/fp_groups.py [hunk 1]
```diff
@@ -184,12 +184,13 @@ def is_complete(self):
     def define(self, alpha, x):
         r"""
         This routine is used in the relator-based strategy of todd coxeter
```
Comment (jksuom) [inline sympy/combinatorics/fp_groups.py]: Preferably Todd-Coxeter with upper case initials.

**Comment by gxyd** [comment]
See #11460

**Pull request closed without merging**